Sage/carbon
# fix: incorrect ref behaviour in Button and IconButton - FE-4924

### Proposed behaviour
Ref should be reset when Button / IconButton gets unmounted

### Current behaviour

Ref value stays the same after the component has been unmounted

### Checklist




#### QA


### Additional context



### Testing instructions

https://codesandbox.io/s/priceless-moser-2fjvm4

Also check that controlled MultiSelect works correctly as my previous implementation of this bug fix broke the way it behaved (Pills were added twice to the list while selecting an Option with an `enter` key)
